azure-iot-device/tests/provisioning/test_sync_provisioning_device_client.py

```diff
@@ -42,31 +49,62 @@ def create_error():
     return RuntimeError("Incoming Failure")
 
 
-class FakePollingMachineSuccess(PollingMachine):
-    def register(self, callback):
-        callback(create_success_result(), error=None)
+def symmetric_key_security_client():
+    return SymmetricKeySecurityClient(
+        provisioning_host=fake_provisioning_host,
+        registration_id=fake_registration_id,
+        id_scope=fake_id_scope,
+        symmetric_key=fake_symmetric_key,
+    )
 
-    def cancel(self, callback):
-        callback()
 
-    def disconnect(self, callback):
-        callback()
+def x509_security_client():
+    mock_x509 = FakeX509(fake_x509_cert_file_value, fake_x509_cert_key_file, fake_pass_phrase)
+    return X509SecurityClient(
+        provisioning_host=fake_provisioning_host,
+        registration_id=fake_registration_id,
+        id_scope=fake_id_scope,
+        x509=mock_x509,
+    )
 
 
+class FakeX509(object):
+    def __init__(self, cert, key, pass_phrase):
+        self.certificate = cert
+        self.key = key
+        self.pass_phrase = pass_phrase
+
+
+different_security_clients = [
+    {
+        "name": "symmetric key security",
+        "class": SymmetricKeySecurityClient,
+        "client_class": SymmetricKeyProvisioningDeviceClient,
+        "security_client": symmetric_key_security_client(),
+    },
+    {
+        "name": "x509 security",
+        "class": X509SecurityClient,
+        "client_class": X509ProvisioningDeviceClient,
+        "security_client": x509_security_client(),
+    },
+]
+
+
+@pytest.mark.parametrize(
+    "params_security_clients",
+    different_security_clients,
+    ids=[x["name"] for x in different_security_clients],
+)
 @pytest.mark.describe("ProvisioningDeviceClient")
 class TestClientCreate:
     xfail_notimplemented = pytest.mark.xfail(raises=NotImplementedError, reason="Unimplemented")
 
     @pytest.fixture
-    def security_client(self):
-        return SymmetricKeySecurityClient(
-            provisioning_host=fake_provisioning_host,
-            registration_id=fake_registration_id,
-            id_scope=fake_id_scope,
-            symmetric_key=fake_symmetric_key,
-        )
+    def security_client(self, params_security_clients):
+        return params_security_clients["security_client"]
 
-    @pytest.mark.it("is created from a symmetric key security client and protocol")
+    @pytest.mark.it("is created from a chosen security client and protocol")
```

Comment:
>is [](start = 21, length = 2)

Nit: capitalize tests

Reply:
I have done some..but i have to go though the tests once again..and check for all of them having proper its..I can do that with the client

---
In reply to: [293477126](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r293477126) [](ancestors = 293477126)